packages/perseus/src/widgets/__tests__/radio/choice_test.jsx

```diff
@@ -0,0 +1,201 @@
+        // Arrange / Act
+        renderChoice();
+
+        const button = screen.getByRole("checkbox", {
+            name: "Select Choice A",
+        });
+
+        // Assert
+        expect(button).toBeVisible();
+    });
+
+    it("has correct aria-checked when checked", () => {
+        // Arrange / Act
+        renderChoice({checked: true});
+
+        const button = screen.getByRole("checkbox", {
+            name: "Select Choice A",
+        });
+
+        // Assert
+        expect(button).toBeChecked();
+    });
+
+    it("has correct aria-checked when unchecked", () => {
+        // Arrange / Act
+        renderChoice({checked: false});
+
+        const button = screen.getByRole("checkbox", {
+            name: "Select Choice A",
+        });
+
+        // Assert
+        expect(button).not.toBeChecked();
+    });
+
+    it("has correct aria-disabled when disabled", () => {
+        // Arrange / Act
+        renderChoice({disabled: true});
+
+        const button = screen.getByRole("checkbox", {
+            name: "Select Choice A",
+        });
+
+        // Assert
+        expect(button.getAttribute("aria-disabled")).toBe("true");
+    });
+
+    it("has correct aria-disabled when not disabled", () => {
+        // Arrange / Act
+        renderChoice({disabled: false});
+
+        const button = screen.getByRole("checkbox", {
+            name: "Select Choice A",
+        });
+
+        // Assert
+        expect(button.getAttribute("aria-disabled")).toBe("false");
+    });
+
+    it("shows correct when in review mode", () => {
+        // Arrange / Act
+        renderChoice({
+            checked: false,
+            correct: true,
+            showCorrectness: true,
+            reviewMode: true,
+        });
+
+        // Assert
+        expect(screen.getByText("Correct")).toBeVisible();
+    });
+
+    it("shows correct selected when in review mode", () => {
+        // Arrange / Act
+        renderChoice({
+            checked: true,
+            correct: true,
+            showCorrectness: true,
+            reviewMode: true,
+        });
+
+        // Assert
+        expect(screen.getByText("Correct (selected)")).toBeVisible();
+    });
+
+    it("shows incorrect when in review mode", () => {
+        // Arrange / Act
+        renderChoice({
+            checked: false,
+            correct: false,
+            showCorrectness: true,
+            reviewMode: true,
+        });
+
+        // Assert
+        expect(screen.getByText("Incorrect")).toBeVisible();
+    });
+
+    it("shows incorrect selected when in review mode", () => {
+        // Arrange / Act
+        renderChoice({
+            checked: true,
+            correct: false,
+            showCorrectness: true,
+            reviewMode: true,
+        });
+
+        // Assert
+        expect(screen.getByText("Incorrect (selected)")).toBeVisible();
+    });
+
+    it("can be checked", () => {
+        // Arrange
+        let checked = false;
```

Comment:
`let` is necessary here because `checked` is updated in the `onChange` handler below.

Reply:
It's because it changes:

``` JS
        let checked = false; // this assigns checked to false
        renderChoice({
            checked,
            onChange: (next) => {
                checked = next.checked; // this reassigns checked to next.checked
            },
        });
```

Semantically `const` suggests something that doesn't change and `let` suggests something that does change. Or more accurately, `const` allows for only one assignment and `let` allows reassignment.

Reply:
Oh wow. I think some of my learning on this was a little misled at some point. Thanks for clarifying. I thought const variables just couldn't change their type not that they couldn't be reassigned at all. Thank you both!